packages/flutter_test/lib/src/accessibility.dart

```diff
@@ -196,131 +234,165 @@ class MinimumTextContrastGuideline extends AccessibilityGuideline {
+
+    final Map<Color, int> colorHistogram = _colorsWithinRect(byteData, paintBounds, image.width, image.height);
+
+    // Node was too far off screen.
+    if (colorHistogram.isEmpty) {
+      return result;
+    }
+
+    final _ContrastReport report = _ContrastReport(colorHistogram);
+
+    final double contrastRatio = report.contrastRatio();
+    final double targetContrastRatio = this.targetContrastRatio(fontSize, isBold: isBold);
+
+    if (contrastRatio - targetContrastRatio >= _tolerance) {
+      return result + const Evaluation.pass();
+    }
+    return result +
+        Evaluation.fail(
+          '$node:\n'
+          'Expected contrast ratio of at least $targetContrastRatio '
+          'but found ${contrastRatio.toStringAsFixed(2)} '
+          'for a font size of $fontSize.\n'
+          'The computed colors was:\n'
+          'light - ${report.lightColor}, dark - ${report.darkColor}\n'
+          'See also: '
+          'https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-contrast.html',
+        );
   }
 
-  // Skip routes which might have labels, and nodes without any text.
-  bool _shouldSkipNode(SemanticsData data) {
-    if (data.hasFlag(ui.SemanticsFlag.scopesRoute))
-      return true;
-    if (data.label.trim().isEmpty && data.value.trim().isEmpty)
-      return true;
-    return false;
+  /// Returns whether node should be skipped.
+  ///
+  /// Skip routes which might have labels, and nodes without any text.
+  bool shouldSkipNode(SemanticsData data) =>
+      data.hasFlag(ui.SemanticsFlag.scopesRoute) ||
+      (data.label.trim().isEmpty && data.value.trim().isEmpty);
+
+  /// Returns if a rectangle of node is off the screen.
+  ///
+  /// Allows node to be of screen partially before culling the node.
+  bool isNodeOffScreen(Rect paintBounds, ui.FlutterView window) {
+    final Size windowPhysicalSize = window.physicalSize * window.devicePixelRatio;
+    return paintBounds.top < -50.0 ||
+           paintBounds.left < -50.0 ||
+           paintBounds.bottom > windowPhysicalSize.height + 50.0 ||
+           paintBounds.right > windowPhysicalSize.width + 50.0;
   }
 
-  // Returns a rect that is entirely on screen, or null if it is too far off.
-  //
-  // Given a pixel buffer based on the physical window size, can we actually
-  // get all the data from this node? allow a small delta overlap before
-  // culling the node.
-  bool _isNodeOffScreen(Rect paintBounds, ui.FlutterView window) {
-    return paintBounds.top < -50.0
-      || paintBounds.left <  -50.0
-      || paintBounds.bottom > (window.physicalSize.height * window.devicePixelRatio) + 50.0
-      || paintBounds.right > (window.physicalSize.width * window.devicePixelRatio)  + 50.0;
+  /// Returns required contrast ratio for font size and font weight.
+  ///
+  /// Defined by http://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-contrast.html
+  double targetContrastRatio(double? fontSize, {required bool isBold}) {
```

Comment:
its more idiomatic to call this `bold` here.

```suggestion
  double targetContrastRatio(double? fontSize, {required bool bold}) {
```